tests/ui/coherence/orphan-check-projections-dont-cover.rs

```diff
@@ -0,0 +1,26 @@
+// Regression test for issue #99554.
+// Don't consider projections to cover type parameters.
+
+// revisions: classic next
+//[next] compile-flags: -Ztrait-solver=next
+
+// aux-crate:foreign=parametrized-trait.rs
+// edition:2021
+
+trait Identity {
+    type Output;
+}
+
+impl<T> Identity for T {
+    type Output = T;
+}
+
+struct Local;
+
+impl<T> foreign::Trait<Local, T, ()> for <T as Identity>::Output {}
```

Comment:
can you explicitly change `orphan_check_trait_ref` to consider aliases which reference type params as uncovered params, otherwise the following test still passes:
```suggestion
struct Local;

impl<T: Identity> foreign::Trait<Local, T, ()> for <T as Identity>::Output {}
```
with a downstream crate proving an impl of `Identity`.